### PR TITLE
pam_ldap: src from GitHub, fix license, add changelog, finalAttrs

### PR DESCRIPTION
**Pull request by lucasew**
- **pam_ldap: src from GitHub, fix license, add changelog, finalAttrs**

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lucasew** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 369276`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 3 tests built:</summary>
  <ul>
    <li>nixosTests.lldap</li>
    <li>nixosTests.openldap</li>
    <li>nixosTests.sssd-ldap</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>pam_ldap</li>
  </ul>
</details>

**Comment by lucasew** [comment]
The difference between the two srcs are the following:

```diff
diff --git a/pam_ldap-186/CVSVersionInfo.txt b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/CVSVersionInfo.txt
index 8e4a520..7356f35 100644
--- a/pam_ldap-186/CVSVersionInfo.txt
+++ b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/CVSVersionInfo.txt
@@ -1,6 +1,6 @@
 # Created and modified by checkpoint; do not edit
-# $Id: CVSVersionInfo.txt,v 1.206 2010/11/08 00:58:42 lukeh Exp $
-# $Name: pam_ldap-186 $
+# $Id$
+# $Name$
 ProjectName: pam_ldap
 ProjectVersion: 186
 ProjectMaintainer: lukeh
diff --git a/pam_ldap-186/ChangeLog b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/ChangeLog
index 41c708a..c0e1199 100644
--- a/pam_ldap-186/ChangeLog
+++ b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/ChangeLog
@@ -1,4 +1,4 @@
-$Id: ChangeLog,v 1.217 2010/11/08 00:55:08 lukeh Exp $
+$Id$
 ===============================================================
 
 186	Luke Howard <lukeh@padlcom>
diff --git a/pam_ldap-186/certutil b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/certutil
index 5ceae89..0d847fa 100644
--- a/pam_ldap-186/certutil
+++ b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/certutil
@@ -1,5 +1,5 @@
 #!/bin/sh
-#ident $Id: certutil,v 1.2 2002/04/14 14:16:16 lukeh Exp $
+#ident $Id$
 #
 # certutil -- manage trusted X.509 certificates
 # inspired by Netscape PKCS #11 toolkit
diff --git a/pam_ldap-186/cvslib.pl b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/cvslib.pl
index e809ea1..1ecf075 100644
--- a/pam_ldap-186/cvslib.pl
+++ b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/cvslib.pl
@@ -1,4 +1,4 @@
-#ident $Id: cvslib.pl,v 1.1 2009/11/06 10:27:49 lukeh Exp $
+#ident $Id$
 
 $CVSVERSIONDIR = $ENV{'CVSVERSIONDIR'};
 
diff --git a/pam_ldap-186/exports.linux b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/exports.linux
index 6c24bc7..dc80d14 100644
--- a/pam_ldap-186/exports.linux
+++ b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/exports.linux
@@ -1,4 +1,4 @@
-#ident $Id: exports.linux,v 1.1 2002/07/31 20:36:37 lukeh Exp $
+#ident $Id$
 EXPORTED {
 	global:
 		# Published PAM service module interfaces
diff --git a/pam_ldap-186/exports.solaris b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/exports.solaris
index fe49c8a..d9c33b6 100644
--- a/pam_ldap-186/exports.solaris
+++ b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/exports.solaris
@@ -1,4 +1,4 @@
-#ident $Id: exports.solaris,v 1.1 2002/07/31 19:47:03 lukeh Exp $
+#ident $Id$
 pam_ldap.so.1 {
 	global:
 		# Published PAM service module interfaces
diff --git a/pam_ldap-186/ldap.conf b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/ldap.conf
index c4522c0..0f43d47 100644
--- a/pam_ldap-186/ldap.conf
+++ b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/ldap.conf
@@ -1,4 +1,4 @@
-# @(#)$Id: ldap.conf,v 1.38 2006/05/15 08:13:31 lukeh Exp $
+# @(#)$Id$
 #
 # This is the configuration file for the LDAP nameservice
 # switch library and the LDAP PAM module.
diff --git a/pam_ldap-186/ldapns.schema b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/ldapns.schema
index 21ae00c..41bbbc7 100644
--- a/pam_ldap-186/ldapns.schema
+++ b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/ldapns.schema
@@ -1,4 +1,4 @@
-# $Id: ldapns.schema,v 1.3 2003/05/29 12:57:29 lukeh Exp $
+# $Id$
 
 # LDAP Name Service Additional Schema
 
diff --git a/pam_ldap-186/md5.c b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/md5.c
index 0b6d729..2eb6b23 100644
--- a/pam_ldap-186/md5.c
+++ b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/md5.c
@@ -21,7 +21,7 @@
   ghost@aladdin.com
 
  */
-/*$Id: md5.c,v 1.4 2002/08/05 03:56:55 lukeh Exp $ */
+/*$Id$ */
 /*
   Independent implementation of MD5 (RFC 1321).
 
diff --git a/pam_ldap-186/md5.h b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/md5.h
index a0ecf71..053a0ee 100644
--- a/pam_ldap-186/md5.h
+++ b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/md5.h
@@ -21,7 +21,7 @@
   ghost@aladdin.com
 
  */
-/*$Id: md5.h,v 1.3 2002/08/05 03:56:55 lukeh Exp $ */
+/*$Id$ */
 /*
   Independent implementation of MD5 (RFC 1321).
 
diff --git a/pam_ldap-186/mkinstalldirs b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/mkinstalldirs
index 5cb6706..6b3b5fc 100755
--- a/pam_ldap-186/mkinstalldirs
+++ b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/mkinstalldirs
@@ -4,7 +4,7 @@
 # Created: 1993-05-16
 # Public domain
 
-# $Id: mkinstalldirs,v 1.2 2000/05/30 08:36:15 lukeh Exp $
+# $Id$
 
 errstatus=0
 
diff --git a/pam_ldap-186/ns-pwd-policy.schema b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/ns-pwd-policy.schema
index 823b721..c5e148d 100644
--- a/pam_ldap-186/ns-pwd-policy.schema
+++ b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/ns-pwd-policy.schema
@@ -1,4 +1,4 @@
-#ident $Id: ns-pwd-policy.schema,v 1.2 2004/03/01 23:47:57 lukeh Exp $
+#ident $Id$
 #
 # Netscape password policy schema
 # draft-vchu-ldap-pwd-policy-00.txt
diff --git a/pam_ldap-186/pam.conf b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/pam.conf
index a687f28..44a4592 100644
--- a/pam_ldap-186/pam.conf
+++ b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/pam.conf
@@ -1,4 +1,4 @@
-#ident $Id: pam.conf,v 1.3 1998/10/05 05:01:13 lukeh Exp $
+#ident $Id$
 #
 # PAM configuration for LDAP is sufficient, otherwise UNIX
 # mandatory authentication policy.
diff --git a/pam_ldap-186/pam_ldap.5 b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/pam_ldap.5
index 0ebcfbf..621041e 100644
--- a/pam_ldap-186/pam_ldap.5
+++ b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/pam_ldap.5
@@ -1,7 +1,7 @@
 .TH pam_ldap 5
 .\" Copyright 1997-2005 Luke Howard."
 .\" Copying restrictions apply. See COPYING.
-.\" $Id: pam_ldap.5,v 1.8 2006/05/15 08:13:31 lukeh Exp $
+.\" $Id$
 .SH NAME
 pam_ldap \- LDAP pluggable authentication module
 .SH DESCRIPTION
diff --git a/pam_ldap-186/pam_ldap.c b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/pam_ldap.c
index 5920d0b..85d801d 100644
--- a/pam_ldap-186/pam_ldap.c
+++ b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/pam_ldap.c
@@ -149,7 +149,7 @@
 #endif
 
 static char rcsid[] __UNUSED__ =
-  "$Id: pam_ldap.c,v 1.215 2010/11/08 00:58:36 lukeh Exp $";
+  "$Id$";
 #if LDAP_SET_REBIND_PROC_ARGS < 3
 static pam_ldap_session_t *global_session = 0;
 #endif
diff --git a/pam_ldap-186/vers_string b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/vers_string
index 52e333b..5a7ac03 100755
--- a/pam_ldap-186/vers_string
+++ b/nix/store/2gd0371np7nwngxiy9l8wyn97g5m2p3d-source/vers_string
@@ -1,6 +1,6 @@
 #!/usr/bin/perl
 #
-#ident $Id: vers_string,v 1.2 2010/02/25 14:40:39 lukeh Exp $
+#ident $Id$
 #
 # Implementation of SGS vers_string which uses CVSVersionInfo.txt.
 
@@ -8,7 +8,7 @@ require "cvslib.pl";
 
 $OUTFILE = (-f "version.h") ? "version.h" : "vers.c";
 $AUTHOR = `whoami`;
-$HEADER = '$Id: vers_string,v 1.2 2010/02/25 14:40:39 lukeh Exp $';
+$HEADER = '$Id$';
 
 if ($ENV{'PROGRAM'}) { $PROGRAM = $ENV{'PROGRAM'}; }
 
```

File: pkgs/by-name/pa/pam_ldap/package.nix [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
 
-  src = fetchurl {
-    url = "https://www.padl.com/download/pam_ldap-${version}.tar.gz";
-    sha256 = "0lv4f7hc02jrd2l3gqxd247qq62z11sp3fafn8lgb8ymb7aj5zn8";
+  src = fetchFromGitHub {
+    owner = "PADL";
+    repo = "pam_ldap";
+    rev = "d1c2e08e4b023708648e2d720add19da6caf77d0";
+    hash = "sha256-W0n01HqWZaEoVShXJyND3xGe+snClGm+GZRw7P4fX10=";
+    # sha256 = "0lv4f7hc02jrd2l3gqxd247qq62z11sp3fafn8lgb8ymb7aj5zn8";
```
[the remark this answers is not in the document]
Comment (lucasew) [reply pkgs/by-name/pa/pam_ldap/package.nix]: done

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
Maybe add `unstableGitUpdater` ?

**Review by GaetanLepage — APPROVED** [review]
LGTM !

**Comment by lucasew** [comment]
Let's see how it goes. It will not post the results here tho.

https://github.com/lucasew/nixcfg/actions/runs/12559999412

**Pull request merged**